scripts/zkai_attention_kv_d16_fused_softmax_table_native_gate.py

```
@@ -0,0 +1,801 @@
+        "source_plus_sidecar_raw_proof_bytes": SOURCE_PLUS_SIDECAR_RAW_PROOF_BYTES,
+        "table_multiplicities": list(TABLE_MULTIPLICITIES),
+        "timing_policy": TIMING_POLICY,
+        "non_claims": list(NON_CLAIMS),
+    }
+
+
+def validate_source_artifacts(source_input: dict[str, Any], source_envelope: dict[str, Any], sidecar_envelope: dict[str, Any]) -> None:
+    try:
+        SOURCE_INPUT_MODULE.validate_payload(source_input)
+    except Exception as err:
+        raise AttentionKvD16FusedSoftmaxTableGateError(f"source input validation drift: {err}") from err
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)